json/parse.go

```diff
@@ -24,6 +24,19 @@ const (
 	quote  = '"'
 )
 
+func internalParseFlags(b []byte) (flags ParseFlags) {
+	// Don't consider surrounding whitespace
+	b = skipSpaces(b)
+	b = b[:len(b)-trailingSpaces(b)]
```

Comment:
There may be an opportunity to remove the bound check here if we change the API to this:
```go
func trimTrailingSpaces(b []byte) []byte {
   ...
}
```
I believe the compiler has to inject bound checks in the current form because it doesn't know that the value returned by `trailingSpaces` will be `<= len(b)`.

Reply:
See 96a44a2.